lua/go.lua

```diff
@@ -15,13 +15,16 @@ function go.setup(cfg)
   vim.cmd('command! Gofmt lua require("go.format").gofmt()')
   vim.cmd('command! Goimport lua require("go.format").goimport()')
 
-  vim.cmd([[command GoBuild :setl makeprg=go\ build | :Gmake]])
-  vim.cmd([[command GoGenerate  :setl makeprg=go\ generate | :Gmake]])
-  vim.cmd([[command GoRun       :setl makeprg=go\ run | :Gmake]])
-  vim.cmd([[command GoTestFunc  :Gmake -run ..]])
 
-  vim.cmd([[command GoTest :setl makeprg=go\ test\ ./... | :Gmake]])
-  vim.cmd([[command GoTestCompile  setl makeprg=go\ build | :Gmake]])
+  local cmds = vim.api.nvim_get_commands({})
+
+  if cmds["GoBuild"]       == nil then vim.cmd([[command GoBuild        :setl makeprg=go\ build | :Gmake]]) end
```

Comment:
if cmds["GoBuild"] == nil then vim.cmd([[command GoBuild :setl makeprg=go\ build | :Gmake]]) end

Or use bang ?
vim.cmd([[command! GoBuild :setl makeprg=go\ build | :Gmake]])

Reply:
Since you did add a bang to the other commands I assumed you purposefully didn't bang these couple of commands, therefore I added the check, but of course adding a bang to these commands would also serve the same purpose.